### ISSUE TITLE
[8.x] No faker formatters in database testing

### ISSUE THREAD

**Issue by fheider** [laravel/framework#34384]
- Laravel Version: 8.2.0
- PHP Version: 7.4.10
- Database Driver & Version: mysql 8

### Description:

The new factories doesnt work in database testing. Error "Unknown formatter". The factory works fine in seeding etc. 

### Steps To Reproduce:

- create model factory and use a definition with faker
- create an unit test
- create an entity with the factory in the unit test
- error "unknown formatter"

**Comment by drashoor** [comment]
same problem

**Comment by sanis** [comment]
Try to extend 

```Tests\TestCase```

or use 

```\Tests\CreatesApplication```

 trait which boots up applications and puts dependencies to the services (including faker providers). 

Standard unit test doesn't have to boot full application. This looks more like functional test to me and that's why it's not working correct way.

**Comment by drashoor** [comment]
> 
> 
> Try to extend
> 
> `\Tests\TestCase`
> 
> or use
> 
> `\Tests\CreatesApplication`
> 
> trait which boots up applications and puts dependencies to the services (including faker providers).
> 
> Standard unit test doesn't have to boot full application. This looks more like functional test to me and that's why it's not working correct way.

Not working

**Comment by taylorotwell** [comment]
Needs to be a feature test.

**Comment by Joakal** [comment]
If anyone is confused. The above issue was a result of this change: https://github.com/laravel/framework/commit/e30a0c979d98f2f1f7b6c565e4002734237a280b

tl;dr: The unit tests no longer use faker by default. It's recommended to use feature tests.

**Comment by cjlaborde** [comment]
Here is the "Real" Solution
Use Tests\TestCase; rather then use PHPUnit\Framework\TestCase;
Laravel ships with the later, but Tests\TestCase class take care of setting up the application, 
otherwise models wont be able to communicate with the database if they are extending PHPunit\Framework\TestCase.

**Comment by fattynomnom** [comment]
I'm using `Tests/TestCase` but I still have this error. Can someone take a look at my code and let me know what I'm doing wrong? 

```
<?php

namespace Tests\Feature;

use App;
use Tests\TestCase;
use App\Models\Attribute;
use Illuminate\Http\Response;

class AttributeTest extends TestCase
{
    protected $attribute;

    public function setUp(): void
    {
        $this->attribute = Attribute::factory()->translated()->make();
    }

    ...
```

**Comment by surgiie** [comment]
> public function setUp(): void
>     {
>         $this->attribute = Attribute::factory()->translated()->make();
>     }

To me it looks like you are not calling the parent setUp method:
```php
    public function setUp(): void
    { 
         parent::setUp();
         // your code
    }

```

**Comment by ProteanCode** [comment]
> Needs to be a feature test.

Please, elaborate on what makes test featured from code perspective.
Is it a child of Tests\TestCase, or being present in Feature directory?

**Comment by ProteanCode** [comment]
Temporary replacement:

`
        $f = \Database\Factories\UserFactory::new();
        $user = $f->createOne();`

**Comment by nickpoulos** [comment]
Having the same issue after upgrading from 7.x to 8.x.  

My tests are extending the proper class.  

```php
<?php

namespace Tests\Unit;

use Tests\TestCase;
use App\Actions\LogCreative;
use App\Http\Dto\CreativeDto;
use Illuminate\Support\Facades\Config;
use Illuminate\Support\Facades\Storage;

class LogCreativeActionTest extends TestCase
{
    protected LogCreative $action;

    public function setUp() : void
    {
        parent::setUp();

        Config::set('logging.log_buffered_creatives', true);
        $this->action = app()->make(LogCreative::class);
    }

    /**
     * Test LogCreative->execute()
     * @dataProvider executeProvider
     * @param CreativeDto $creative
     * @param bool $force
     * @return void
     * @throws \Illuminate\Contracts\Filesystem\FileNotFoundException
     */
    public function testExecute(CreativeDto $creative, bool $force) : void
    {
        Storage::fake('s3');

        $filename = $this->action->execute($creative, $force);

        if ($force) {
            Storage::assertExists($filename);
        } else {
            Storage::assertMissing($filename);
        }
    }

    public function executeProvider() : array
    {
        return [
            "Log Creative With Force Set To False" => [
                CreativeDto::factory()->make(),
                false
            ],
            "Log Creative With Force Set To True" => [
                CreativeDto::factory()->make(),
                true
            ]
        ];
    }
}

```

**Comment by rvanbaalen** [comment]
@surgiie's suggestion was the one for me; I failed to call parent::setUp() in my custom setUp method. When I added that, it all worked again.
